src/main/java/org/elasticsearch/index/query/TermQuery.java

```diff
@@ -0,0 +1,205 @@
+        out.writeFloat(boost);
+        out.writeOptionalString(queryName);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        if (boost == 1.0f && queryName == null) {
+            builder.field(fieldName, value);
+        } else {
+            builder.startObject(fieldName);
+            builder.field("value", value);
+            if (boost != 1.0f) {
+                builder.field("boost", boost);
+            }
+            if (queryName != null) {
+                builder.field("_name", queryName);
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+    }
+
+    @Override
+    public String[] names() {
+        return new String[]{TermQuery.NAME};
+    }
+
+    @Override
+    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        TermQuery query = new TermQuery();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public void fromXContent(QueryParseContext context) throws IOException {
+        XContentParser parser = context.parser();
+        XContentParser.Token token = parser.nextToken();
+        if (token != XContentParser.Token.FIELD_NAME) {
+            throw new QueryParsingException(context.index(), "[term] query malformed, no field");
+        }
+        fieldName = parser.currentName();
+        queryName = null;
+        value = null;
+        boost = 1.0f;
+        token = parser.nextToken();
+        if (token == XContentParser.Token.START_OBJECT) {
+            String currentFieldName = null;
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    currentFieldName = parser.currentName();
+                } else {
+                    if ("term".equals(currentFieldName)) {
+                        value = parser.objectBytes();
+                    } else if ("value".equals(currentFieldName)) {
+                        value = parser.objectBytes();
+                    } else if ("boost".equals(currentFieldName)) {
+                        boost = parser.floatValue();
+                    } else if ("_name".equals(currentFieldName)) {
+                        queryName = parser.text();
+                    } else {
+                        throw new QueryParsingException(context.index(), "[term] query does not support [" + currentFieldName + "]");
+                    }
+                }
+            }
+            parser.nextToken();
+        } else {
+            value = parser.text();
+            // move to the next token
+            parser.nextToken();
+        }
+    }
+
+    /**
+     * Produces a lucene query from this elasticsearch query
+     */
+    public Query toQuery(QueryParseContext parseContext) {
+        if (value == null) {
```

Comment:
I think we should also check if `fieldName` is null and fail? maybe use Preconditions here for simplicity?